### ISSUE TITLE
[FEATURE REQUEST] WeakAura or plater script that reads a route and paints an indicator on name plates of mobs required by the route

### ISSUE THREAD

**Issue by jmartin84** [Tomslack/ManbabyDungeonTools#40]
It would be pretty awesome to do some sort of integration with wa or plater to give people a visual indicator mid dungeon as to what trash is included in a route. Could eventually be extended to additional meta data like 'this pack will trigger prideful' etc etc

**Comment by srenauld** [comment]
Hey!

I need to have a proper look at how, but I'm not saying no. It's something that could help a lot of new tanks out, for instance - map the route then auto-mark as you go along.

**Comment by Esteth** [comment]
Blizzard has made this "Impossible" deliberately. It's not possible to tell mobs with the same name apart from each other throughout the dungeon, so if you have a "mighty monster" you have to kill, and another "mighty monster" you don't, then there's no way for the addon to highlight one but not the other.

**Comment by psykzz** [comment]
Just curious for this, but does the UnitGUID not help or is that regenerated for each run?

**Comment by Esteth** [comment]
UnitGUID is regenerated for each spawn of each monster, so you can't rely on "the right guy in the 3rd pack" having the same UnitGUID from run to run.

**Comment by malachantrio** [comment]
Not sure if this would work (especially for patrolling mobs) but one possible option might be to have a list of the mobs names in the next pack to be pulled somewhere on the screen with an arrow pointing at the coords that those mobs are located. Wouldn't mark the pack but would give a tank the ability to quickly work out what's next without bringing up the map. It would also need to be able to understand when a pull has been completed which again I'm not sure how that could be implemented, but just thought I'd throw the idea out there

**Comment by justinryder** [comment]
> It would also need to be able to understand when a pull has been completed which again I'm not sure how that could be implemented

If this display went under the assumption that the order of pulls lines up with the order you actually run the dungeon (requiring the plan to be split into distinct pulls would help in this regard, but order added to the plan could also work), when a mob is pulled, it could make the assumption that the first match by name in the planned pull list is that monster and mark it as pulled, then eventually killed.

A naive implementation that would still be pretty handy could even just be as simple as a checklist:
* 3/10 Spriggens (1.5/5%)
* 4/22 Runestags (8/44%)
etc

**Comment by srenauld** [comment]
@justinryder There would be quite a bit of value in that, particularly since it also would work as a good visual reminder. We ruined a HoA key earlier because we all forgot to pull the dredgers before the first shard to make all the pridefuls align, for instance - a visual reminder like that would've likely saved that.

**Comment by justinryder** [comment]
This has been the feature I've always dreamed of having in MDT, but I've never known how to go about making it work. I usually resort to screenshotting the route and having that open in another window until I've memorized the path, which is...less than ideal.

I'm wondering if there would be a good way to expose the route data so that multiple addons could access it, which would allow for many different ways of displaying the same info (native/WA/plater/elvui/3rd party/etc). Sort of like how TSM data is read and displayed by many different addons in lots of different ways.

I'd be very interested in helping develop this feature, but I'm still quite the newbie to Lua and the WoW APIs (been learning through custom WAs) - JS/web dev is my strength.

**Comment by srenauld** [comment]
@justinryder @travisneids was messing around with a Plater-DBM integration today, with the intention of figuring out what is available/possible and what isn't.

We can definitely build support so other add-ons can pull route data in a format that is easy to understand; like, right now, a pitfall of the route format is that it relies on a lot of "hidden" data (coordinates aren't WoW map coordinates but MDT-specific, groups are opaque and hard to parse through, the data as to what would be worth assigning - kicks, CCs, whatever - just doesn't exist), so that would need to be sorted out in order to do this properly, but I totally agree with you: even if we can only get "You're pulling 2 Soulcleaver, 2 harvesters; assign kicks for Harvesters and run away from Soulcleavers during hand of thros", that'd already be a _gigantic_ win for people coming to M+

**Comment by gxlinx** [comment]
I wonder if it would be easier to create path markers for addons like tom tom or something similar that points you in the correct direction with a small description. like Pull left pack, CC mob run past, etc, etc. They can use the UI to determine which is the correct pack to pull if things are confusing.  

I wonder if it would be better if this add-on overlaid the actual map. Then you don't have to type the command and find the area you are in. Probably require a full rewrite though.

**Comment by chalkos** [comment]
There's an addon that already does (or did, as it got broken in one of the recent MDT versions) something towards this: https://www.curseforge.com/wow/addons/mdt-guide

It reduces the MDT window to show only map and pulls, and keeps focusing the next pull according to the current % (both on the list and on the map). It is a simple implementation but I'm not sure the wow api can provide much more...
